### ISSUE TITLE
Null Object Reference crash deleting dragon whose Fire Ball attack is mid-air

### ISSUE THREAD

**Issue by IBwWG** [larsiusprime/tdrpg-bugs#801]
0.7.8, win7x64, HD mode@1704x960windowed

Was just coming from #694 (Bonus 1 on NG+) and thought, well, I'm not going to finish, so let's see if deleting some defenders helps.  Next time it unfroze I deleted 3 defenders, and then it crashed:

```
...
FlxUIButton name = bonus_btn
AL lib: (EE) MMDevApiProc: WaitForSingleObjectEx error: 0x102
Called from ApplicationMain::main ApplicationMain.hx line 159
Called from ApplicationMain::create ApplicationMain.hx line 71
Called from lime.app.Application::exec lime/app/Application.hx line 222
Called from lime._backend.native.NativeApplication::exec lime/_backend/native/Na
tiveApplication.hx line 136
Called from lime._backend.native.NativeApplication::handleRenderEvent lime/_back
end/native/NativeApplication.hx line 368
Called from lime.app.Event_Void_Void::dispatch lime/app/Event.hx line 145
Called from *::_Function_2_1 lime/app/Application.hx line 110
Called from lime.app.Application::render lime/app/Application.hx line 762
Called from openfl.display.Stage::render openfl/display/Stage.hx line 761
Called from openfl.display.DisplayObjectContainer::__broadcast openfl/display/Di
splayObjectContainer.hx line 355
Called from openfl.display.DisplayObject::__broadcast openfl/display/DisplayObje
ct.hx line 251
Called from openfl.events.EventDispatcher::__dispatchEvent openfl/events/EventDi
spatcher.hx line 237
Called from flixel.FlxGame::onEnterFrame flixel/FlxGame.hx line 530
Called from flixel.FlxGame::step flixel/FlxGame.hx line 663
Called from com.leveluplabs.tdrpg.Main::update com/leveluplabs/tdrpg/Main.hx lin
e 1495
Called from flixel.FlxGame::update flixel/FlxGame.hx line 752
Called from flixel.FlxState::tryUpdate flixel/FlxState.hx line 168
Called from com.leveluplabs.tdrpg.State_Battle::update com/leveluplabs/tdrpg/Sta
te_Battle.hx line 905
Called from com.leveluplabs.tdrpg.State_Battle::battleUpdate com/leveluplabs/tdr
pg/State_Battle.hx line 1020
Called from com.leveluplabs.tdrpg.BattleModel::update com/leveluplabs/tdrpg/Batt
leModel.hx line 1417
Called from com.leveluplabs.tdrpg.BattleModel::doUpdate com/leveluplabs/tdrpg/Ba
ttleModel.hx line 1825
Called from com.leveluplabs.tdrpg.BattleModel::updateProjectiles com/leveluplabs
/tdrpg/BattleModel.hx line 3308
Called from com.leveluplabs.tdrpg.ProjectileSprite::updateFromModel com/levelupl
abs/tdrpg/ProjectileSprite.hx line 166
Called from com.leveluplabs.tdrpg.ProjectileSprite::calcPath com/leveluplabs/tdr
pg/ProjectileSprite.hx line 642
Called from com.leveluplabs.tdrpg.BattleModel::forceProjCollision com/leveluplab
s/tdrpg/BattleModel.hx line 599
Called from com.leveluplabs.tdrpg.BattleModel::projectileCollision com/levelupla
bs/tdrpg/BattleModel.hx line 2895
Called from com.leveluplabs.tdrpg.CreatureSprite::hitByProjectile com/leveluplab
s/tdrpg/CreatureSprite.hx line 2047
Called from com.leveluplabs.tdrpg.BattleModel::fireProjectileSplash com/levelupl
abs/tdrpg/BattleModel.hx line 243
Called from com.leveluplabs.tdrpg.BattleModel::fireSplash com/leveluplabs/tdrpg/
BattleModel.hx line 416
Called from com.leveluplabs.tdrpg.Zone::copy com/leveluplabs/tdrpg/Zone.hx line
44
Called from com.leveluplabs.tdrpg.Zone::new com/leveluplabs/tdrpg/Zone.hx line 3
8
Called from com.leveluplabs.tdrpg.Zone::set_ptr_owner com/leveluplabs/tdrpg/Zone
.hx line 95
Called from com.leveluplabs.tdrpg.Zone::refreshValuesFromOwner com/leveluplabs/t
drpg/Zone.hx line 69
AL lib: (EE) alc_cleanup: 1 device not closed
```

(I tried to take a screenshot, but unfortunately I was trying to include the task manager showing mem usage--around 1.1 GB for DQ--and after arranging the windows appropriately, the main game window was then black, so then it was too late.)

**Comment by IBwWG** [comment]
Reproduced on a normal level (extreme, NG+):  Part way through the battle, deleted 3-4 defenders, and pow:

![dq-crash-deleting-defender](https://cloud.githubusercontent.com/assets/16719964/15692074/af15cb74-278c-11e6-8b6c-b9a3ca3070f7.png)


```
FlxUIButton name = bonus_btn
Called from ApplicationMain::main ApplicationMain.hx line 159
Called from ApplicationMain::create ApplicationMain.hx line 71
Called from lime.app.Application::exec lime/app/Application.hx line 222
Called from lime._backend.native.NativeApplication::exec lime/_backend/native/Na
tiveApplication.hx line 136
Called from lime._backend.native.NativeApplication::handleRenderEvent lime/_back
end/native/NativeApplication.hx line 368
Called from lime.app.Event_Void_Void::dispatch lime/app/Event.hx line 145
Called from *::_Function_2_1 lime/app/Application.hx line 110
Called from lime.app.Application::render lime/app/Application.hx line 762
Called from openfl.display.Stage::render openfl/display/Stage.hx line 761
Called from openfl.display.DisplayObjectContainer::__broadcast openfl/display/Di
splayObjectContainer.hx line 355
Called from openfl.display.DisplayObject::__broadcast openfl/display/DisplayObje
ct.hx line 251
Called from openfl.events.EventDispatcher::__dispatchEvent openfl/events/EventDi
spatcher.hx line 237
Called from flixel.FlxGame::onEnterFrame flixel/FlxGame.hx line 530
Called from flixel.FlxGame::step flixel/FlxGame.hx line 663
Called from com.leveluplabs.tdrpg.Main::update com/leveluplabs/tdrpg/Main.hx lin
e 1495
Called from flixel.FlxGame::update flixel/FlxGame.hx line 752
Called from flixel.FlxState::tryUpdate flixel/FlxState.hx line 168
Called from com.leveluplabs.tdrpg.State_Battle::update com/leveluplabs/tdrpg/Sta
te_Battle.hx line 905
Called from com.leveluplabs.tdrpg.State_Battle::battleUpdate com/leveluplabs/tdr
pg/State_Battle.hx line 1020
Called from com.leveluplabs.tdrpg.BattleModel::update com/leveluplabs/tdrpg/Batt
leModel.hx line 1417
Called from com.leveluplabs.tdrpg.BattleModel::doUpdate com/leveluplabs/tdrpg/Ba
ttleModel.hx line 1827
Called from com.leveluplabs.tdrpg.BattleModel::updateCollisions com/leveluplabs/
tdrpg/BattleModel.hx line 3080
Called from com.leveluplabs.tdrpg.BattleModel::updateProjectileCollisions com/le
veluplabs/tdrpg/BattleModel.hx line 3131
Called from com.leveluplabs.tdrpg.BattleModel::projectileCollision com/levelupla
bs/tdrpg/BattleModel.hx line 2901
Called from com.leveluplabs.tdrpg.CreatureSprite::hitByProjectile com/leveluplab
s/tdrpg/CreatureSprite.hx line 2047
Called from com.leveluplabs.tdrpg.BattleModel::fireProjectileSplash com/levelupl
abs/tdrpg/BattleModel.hx line 243
Called from com.leveluplabs.tdrpg.BattleModel::fireSplash com/leveluplabs/tdrpg/
BattleModel.hx line 416
Called from com.leveluplabs.tdrpg.Zone::copy com/leveluplabs/tdrpg/Zone.hx line
44
Called from com.leveluplabs.tdrpg.Zone::new com/leveluplabs/tdrpg/Zone.hx line 3
8
Called from com.leveluplabs.tdrpg.Zone::set_ptr_owner com/leveluplabs/tdrpg/Zone
.hx line 95
Called from com.leveluplabs.tdrpg.Zone::refreshValuesFromOwner com/leveluplabs/t
drpg/Zone.hx line 69
AL lib: (EE) alc_cleanup: 1 device not closed
```

**Comment by IBwWG** [comment]
For the starting layout, so you can delete also: [export-crash-deleting-defenders.zip](https://github.com/larsiusprime/tdrpg-bugs/files/292276/export-crash-deleting-defenders.zip)

**Comment by IBwWG** [comment]
Maybe it's from deleting a dragon whose fire is still in mid-air?

**Comment by IBwWG** [comment]
Yep, that's it.  Steps to repro:

1. place a dragon
1. boost to level 5
1. disable first 4 levels
1. wait for Fire Ball attack to occur
1. delete before the fireball hits
1. crash when the fireball hits (same stack trace as the above two)

**Comment by larsiusprime** [comment]
Excellent work. Should have this fixed soon.

**Comment by larsiusprime** [comment]
Fixed in 0.8.1
